File: src/com/leozhi/topic912/Solution03.java
package com.leozhi.topic912;

/**
 * @author leozhi
 * 插入排序
 * 通过
 * 1465ms
 */
public class Solution03 {
    public int[] sortArray(int[] nums) {
        int temp;
        for (int i = 1; i < nums.length; i++) {
            for (int j = i; j > 0; j--) {
                if (nums[j] < nums[j - 1]) {
                    temp = nums[j];
                    nums[j] = nums[j - 1];
                    nums[j - 1] = temp;
                }
            }
        }
        return nums;
    }
}

File: src/com/leozhi/topic189/Solution03.kt
package com.leozhi.topic189

/**
 * @author leozhi
 * 数组拷贝
 * 通过
 * 320ms
 */
class Solution03 {
    fun rotate(nums: IntArray, k: Int) {
        val len = nums.size
        for (i in 0 until k) {
            val temp = nums[len - 1]
            System.arraycopy(nums, 0, nums, 1, len - 1)
            nums[0] = temp
        }
    }
}